DatabaseConnection.java

```diff
@@ -0,0 +1,47 @@
+package com.example.demo.repository;
+
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Controller;
+
+import java.sql.*;
+
+@Component
+public class DatabaseConnection {
+
+    // JDBC URL, username and password of MySQL server
+    @Value("${spring.datasource.url}")
+    private String url;
+    @Value("${spring.datasource.username}")
+    private  String user;
+    @Value("${spring.datasource.driver-class-name}")
+    private String driver;
+    @Value("${spring.datasource.password}")
+    private String password;
+
+    // JDBC variables for opening and managing connection
+    private static Connection con;
+    private static Statement stmt;
```

Comment:
1) Unused variables.
2) Do not declare such class fields (Statement, ResultSet). These variables should be created on demand instead.